thingsvision/core/extraction/torch.py

```diff
@@ -91,6 +67,26 @@ def register_hook(self, module_name: str) -> None:
     def _unregister_hook(self) -> None:
         self.hook_handle.remove()
 
+    def extract_batch(
+        self,
+        batch: TensorType["b", "c", "h", "w"],
+        module_name: str,
+        flatten_acts: bool,
+        output_type: str = "tensor",
+    ) -> Union[
+        TensorType["b", "num_maps", "h_prime", "w_prime"],
+        TensorType["b", "t", "d"],
+        TensorType["b", "p"],
+        TensorType["b", "d"],
+    ]:
+        self._module_and_output_check(module_name, output_type)
+        self.register_hook(module_name=module_name)
+        act = self._extract_batch(batch, module_name, flatten_acts)
+        if output_type == "ndarray":
+            act = self._to_numpy(act)
+        self._unregister_hook()
+        return act
+
```

Comment:
@LukasMut did you measure how much overhead does self.register_hook(module_name=module_name) and self._unregister_hook makes?

Reply:
Also `self._module_and_output_check(module_name, output_type)`

Reply:
I tested the overhead based on the extraction of a `resnet50` and there is basically no overhead. For 100 iterations with batch size 64 it took and 34.39s without the register and checks and 34.78s with the new implementation.

Reply:
Thanks for checking. I will leave it as is then.